### PR TITLE
enabling CRT for consul-terraform-sync

### PR DESCRIPTION
**Pull request by 99**
**Enabling the CRT CD pipeline for CTS.**

Final "green" pipeline is [here](https://github.com/hashicorp/consul-terraform-sync/actions/runs/1186508858) 

This includes files and logic to build all binaries, docker images and rpms/debs and attach to the GitHub workflow.

The [promotion](https://github.com/hashicorp/crt-workflows-common/actions) pipeline will then kick off to sign/notarize and move these artifacts to the [stable](https://artifactory.hashicorp.engineering/ui/repos/tree/General/hashicorp-crt-stable-local%2Fconsul-terraform-sync%2F0.3.0%2Fc6c834563b4db401645738d7b4a3023a26a158bd) repo in Artifactory ready to be released.

We should have 25 artifacts in total:

consul-terraform-sync-0.3.0.arm64.rpm
consul-terraform-sync-0.3.0.x86_64.rpm
consul-terraform-sync-0.3.0.armelv5.rpm
consul-terraform-sync-0.3.0.armhfv6.rpm
consul-terraform-sync-0.3.0.i386.rpm

consul-terraform-sync_0.3.0_amd64.deb
consul-terraform-sync_0.3.0_armelv5.deb
consul-terraform-sync_0.3.0_armhfv6.deb
consul-terraform-sync_0.3.0_arm64.deb
consul-terraform-sync_0.3.0_i386.deb

consul-terraform-sync_0.3.0_docker_linux_386.tar
consul-terraform-sync_0.3.0_docker_linux_amd64.tar
consul-terraform-sync_0.3.0_docker_linux_arm.tar 
consul-terraform-sync_0.3.0_docker_linux_arm64.tar

consul-terraform-sync_0.3.0_freebsd_386.zip
consul-terraform-sync_0.3.0_linux_386.zip
consul-terraform-sync_0.3.0_windows_386.zip
consul-terraform-sync_0.3.0_darwin_amd64.zip
consul-terraform-sync_0.3.0_freebsd_amd64.zip
consul-terraform-sync_0.3.0_linux_amd64.zip
consul-terraform-sync_0.3.0_linux_arm64.zip
consul-terraform-sync_0.3.0_linux_armelv5.zip
consul-terraform-sync_0.3.0_linux_armhfv6.zip
consul-terraform-sync_0.3.0_solaris_amd64.zip
consul-terraform-sync_0.3.0_windows_amd64.zip

### CODE REVIEW

File: .release/ci.hcl [hunk 1]
```diff
@@ -0,0 +1,110 @@
+schema = "1"
+
+project "consul-terraform-sync" {
+  team = "consul api tooling"
+  slack {
+    notification_channel = "#feed-releng" #TODO update slack channel
```
Comment (lornasong) [inline .release/ci.hcl:6]: Was wondering if you recommend us switching back to our notification channel after this PR? Want to make sure if you have any plans monitoring this in #feed-releng

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,371 @@
+name: build
+
+on:
+  push
```
Comment (lornasong) [inline .github/workflows/build.yml:4]: Wondering if this means that the CRT GH actions will be triggered every time we push a branch? Just want to double-check my understanding on the process to start the CRT when we're ready for release.

**Review by lornasong — APPROVED** [review]
Thanks so much for all of y'all work on this 🙏! I'm not really familiar with this area of work but reading through it makes sense to me. Fun to see the GitHub Actions in action :). Left a couple of questions to double-check myself. Thank you

File: .release/ci.hcl (hunk 1 above)
Comment (sarahethompson) [reply .release/ci.hcl:6]: Hi @lornasong! Right now, our tooling will only send notifications to #feed-releng but we will be working very soon on getting this enabled for individual team channels and will then circle back and update this value in the CTS OSS/ENT repos 👍

File: .github/workflows/build.yml (hunk 2 above)
Comment (sarahethompson) [reply .github/workflows/build.yml:4]: Yes, the build workflow will be kicked off on every push, and then depending on what's been set [here](https://github.com/hashicorp/consul-terraform-sync/pull/369/files#diff-d51ad13228611273f3804158eadd6b38eb7fe8eb46d5e84b94e6416a15c63709R11) as a release branch, the promotion pipeline will kick off after a successful build workflow on that branch (in [crt-common-workflows])(https://github.com/hashicorp/crt-workflows-common/actions). We are working on an onboarding guide at the moment and will share that with the team once it's ready. We'll also be scheduling a knowledge share session soon with the consul api team and we'll make sure to answer these questions more fully!

Comment (lornasong) [reply .github/workflows/build.yml:4]: Gotcha! Thanks for clarifying and pointing out where the release branch is set. We're planning to release today so will be sure to update that field. Looking forward to the session and on boarding guide 💞

File: .release/ci.hcl (hunk 1 above)
Comment (lornasong) [reply .release/ci.hcl:6]: Hello @sarahethompson! That sounds good! Thanks for clarifying :)

**Pull request merged**